apps/bitwarden_mailconfig.sh

```diff
@@ -58,125 +58,125 @@ then
+
+if [ -z "$PROTOCOL" ]
+then
+    exit 1
+fi
+
+case "$PROTOCOL" in
+    "SSL")
+        DEFAULT_PORT=465
+    ;;
+    "STARTTLS")
+        DEFAULT_PORT=587
+    ;;
+    "NO-ENCRYPTION")
+        DEFAULT_PORT=25
+    ;;
+    *)
+    ;;
+esac
+
+# Enter custom port or just use the default port
+SMTP_PORT=$(whiptail --title "$TITLE" --nocancel --menu \
+"Based on your selection of encryption the default port is $DEFAULT_PORT. Would you like to use that port or something else?
+$MENU_GUIDE\n\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+"Use default port" "($DEFAULT_PORT)" \
+"Enter another port" ""  3>&1 1>&2 2>&3)
+
+if [ -z "$SMTP_PORT" ]
+then
+    exit 1
+fi
+
+case "$SMTP_PORT" in
+    "Use default port")
+        SMTP_PORT="$DEFAULT_PORT"
+    ;;
+    "Enter another port")
+        SMTP_PORT="$(input_box_flow 'Please enter the port for your SMTP Relay.')"
+    ;;
+    *)
+    ;;
+esac
+
+# Enter your SMTP username
+if yesno_box_yes "Does $MAIL_SERVER require any credentials, like username and password?"
+then
+    MAIL_USERNAME=$(input_box_flow "Please enter the SMTP username to your email provider.\nE.g. you@mail.com")
+
+    # Enter your mail user password
+    MAIL_PASSWORD=$(input_box_flow "Please enter the SMTP password to your email provider.")
+fi
+
 while :
 do
-    PROTOCOL=$(input_box "Please type in the encryption protocol for your mailserver.
-The available options are 'SSL', 'STARTTLS' or 'none'.\n\nIf you don't want to change the protocol \
-setting, that are already configured inside the global.override.env-file, just leave the box empty.")
-    if ! yesno_box_yes "Is this correct? $PROTOCOL"
+    ADMIN_ACCOUNT=$(input_box "Please enter mail accounts, that should have access \
+to the Bitwarden admin-panel, reachable under https://your-bitwarden-domain/admin/.
+They don't have to be registered Bitwarden accounts.
+To make this setting work, your Bitwarden mailserver settings have to be correct.
+You can enter just one e-mail address or enter more than one like so:
+'bitwarden@example.com,bitwarden2@example1.com,bitwarden3@example2.com'
+If you want to keep the admin accounts that are already configured inside the \
+global.override.env-file, just leave the box empty.")
+
+    if [ -n "$ADMIN_ACCOUNT" ]
     then
-        msg_box "OK, please try again."
-    else
-        if [ "$PROTOCOL" = "SSL" ]
-        then
-            DEFAULT_PORT=465
-            break
-        elif [ "$PROTOCOL" = "none" ]
+        if yesno_box_yes "Does this look correct: $ADMIN_ACCOUNT"
         then
-            DEFAULT_PORT=25
             break
-        elif [ "$PROTOCOL" = "STARTTLS" ]
-        then
-            DEFAULT_PORT=587
-            break
-        elif [ "$PROTOCOL" = "" ]
-        then
-            DEFAULT_PORT=""
-            break
-        else
-            msg_box "The answer wasn't correct. Please type in 'SSL', 'STARTTLS', 'none' or leave the input box empty."
         fi
+    else
+        break
     fi
 done
```

Comment:
Do we really want allow to leave this empty? Otherwise I would just use input_box_flow here and remove the option to leave this empty.

Reply:
Yeah, let's do it like this.